src/api/svg.ts

```diff
@@ -0,0 +1,537 @@
+    if (widthRaw.includes('%')) {
+      width = (box.width || 0) * (width / 100.0);
+    }
+  } else if (box.width) {
+    width = box.width;
+  }
+
+  let height: number | undefined;
+  if (heightRaw && parseFloat(heightRaw)) {
+    height = parseFloat(heightRaw);
+    if (heightRaw.includes('%')) {
+      height = (box.height || 0) * (height / 100.0);
+    }
+  } else if (box.height) {
+    height = box.height;
+  }
+
+  let x = typeof attributes.x !== 'undefined' ? parseFloat(attributes.x) : 0;
+  let y = typeof attributes.y !== 'undefined' ? parseFloat(attributes.y) : 0;
+
+  let x1 = typeof attributes.x !== 'undefined' ? parseFloat(attributes.x1) : 0;
+  let y1 = typeof attributes.y !== 'undefined' ? parseFloat(attributes.y1) : 0;
+
+  let x2 = typeof attributes.x !== 'undefined' ? parseFloat(attributes.x2) : 0;
+  let y2 = typeof attributes.y !== 'undefined' ? parseFloat(attributes.y2) : 0;
+
+  let cx = typeof attributes.cx !== 'undefined' ? parseFloat(attributes.cx) : 0;
+  let cy = typeof attributes.cy !== 'undefined' ? parseFloat(attributes.cy) : 0;
+
+  let r = typeof attributes.r !== 'undefined' ? parseFloat(attributes.r) : 0;
+  let rx = typeof attributes.rx !== 'undefined' ? parseFloat(attributes.rx) : 0;
+  let ry = typeof attributes.ry !== 'undefined' ? parseFloat(attributes.ry) : 0;
+
+  const fill = fillRaw || box.fill;
+  const fillOpacity =
+    typeof fillOpacityRaw !== 'undefined'
+      ? parseFloat(fillOpacityRaw)
+      : box.fillOpacity;
+  const stroke = strokeRaw || box.stroke;
+  const strokeOpacity =
+    typeof strokeOpacityRaw !== 'undefined'
+      ? parseFloat(strokeOpacityRaw)
+      : box.strokeOpacity;
+  const strokeLineCap = StrokeLineCapMap[strokeLineCapRaw] || box.strokeLineCap;
+  const strokeLineJoin = StrokeLineJoinMap[strokeLineJoinRaw] || box.strokeLineJoin;
+  const strokeWidth =
+    typeof strokeWidthRaw !== 'undefined'
+      ? parseFloat(strokeWidthRaw)
+      : box.strokeWidth;
+  const fontFamily = fontFamilyRaw || box.fontFamily;
+  const fontSize =
+    typeof fontSizeRaw !== 'undefined' ? parseFloat(fontSizeRaw) : box.fontSize;
+
+  if (attributes.transform) {
+    const regexTransform = /(\w+)\((.+?)\)/g;
+    let parsed = regexTransform.exec(attributes.transform);
+    while (parsed !== null) {
+      const [, name, rawArgs] = parsed;
+      const args = (rawArgs || '')
+        .split(/\s*,\s*|\s+/)
+        .filter((value) => value.length > 0)
+        .map((value) => parseFloat(value));
+
+      const result = transform({ x, y }, name, args);
+      x = result.x;
+      y = result.y;
+
+      parsed = regexTransform.exec(attributes.transform);
+    }
+  }
+
+  if (box.converter) {
+    x = box.converter.x(x);
+    y = box.converter.y(y);
+
+    x1 = box.converter.x(x1);
+    y1 = box.converter.y(y1);
+
+    x2 = box.converter.x(x2);
+    y2 = box.converter.y(y2);
+
+    cx = box.converter.x(cx);
+    cy = box.converter.y(cy);
+
+    r = box.converter.x(r);
```

Comment:
`r` should be converted into `const { width, height } = box.converter.size(rx || r, ry || r); rx = width; ry = height;`.